### PR TITLE
adds layout-merging utility

### PR DESCRIPTION
**Pull request by tyarkoni**
Adds a small utility for merging layouts. WIP--still requires a test or two.

### CODE REVIEW

File: grabbit/core.py [hunk 1]
```diff
@@ -573,3 +573,28 @@ def count_matches(f):
...
+        then the result will be updated with values from the 3rd Layout, etc.
+        This means that order matters: in the event of entity or filename
+        conflicts, later layouts will take precedence.
+    '''
+    for l in layouts[1:]:
+        layouts[0].files.update(l.files)
+
+        for k, v in l.entities.items():
+            if k not in layouts[0].entities:
+                layouts[0].entities.update(v)
```
Comment (effigies) [inline grabbit/core.py]: Pretty sure this should be:

```Python
layouts[0].entities[k] = v
```

File: grabbit/core.py [hunk 2]
```diff
@@ -573,3 +573,28 @@ def count_matches(f):
...
+        A BIDSLayout containing merged files and entities.
+    Notes:
+        Layouts will be merged in the order of the elements in the list. I.e.,
+        the first Layout will be updated with all values in the 2nd Layout,
+        then the result will be updated with values from the 3rd Layout, etc.
+        This means that order matters: in the event of entity or filename
+        conflicts, later layouts will take precedence.
+    '''
+    for l in layouts[1:]:
+        layouts[0].files.update(l.files)
```
Comment (effigies) [inline grabbit/core.py]: Does it make sense to leave modify the inputs and instead do:

```Python
layout = layouts[0].clone()  # or copy()?

for l in layouts[1:]:
    layout.files.update(l.files)
    ...
```

I imagine most of the time people aren't going to merge and still want the original untouched, but the effect on the inputs is order-dependent, which feel wrong to me.

**Comment by tyarkoni** [comment]
Good call on both counts. The only issue with cloning the first `Layout` is that I don't think I have a `clone()` method implemented at the moment. Unfortunately deep copying with `deepcopy` won't work out of the box because of the internally-stored regular expressions (and possibly other attributes), so this will require implementing a `__deepcopy__` method, I think.

**Comment by tyarkoni** [comment]
Just pushed it anyway--but will require `__deepcopy__` implementation before it works.

**Comment by tyarkoni** [comment]
Okay, deep copying via a `.clone()` method should work now. I also added a metaclass for `Layout` that enables implicit layout merging by passing multiple paths as a list in the `path` argument.

**Comment by tyarkoni** [comment]
@effigies, let me know if you have a change to look at this again. Changes are pretty minimal, so if you don't have time, no worries--I'll merge as-is.

**Comment by effigies** [comment]
Nothing jumps out at me, but I'll admit I didn't look at the tests at all. If you're comfortable with this, I'm happy to see it merged.

**Pull request merged**